### PR TITLE
A4A: Implement the payment method overview page.

### PR DESCRIPTION
**Pull request by jkguidaven**
This ported the payment method list from Jetpack Manage to the A4A portal. Note that the data is temporarily mocked to demonstrate the UI while awaiting a working backend API.

**Empty State**
Screenshot 2024-03-05 at 4 15 54 PM
**Populated State**
Screenshot 2024-03-05 at 4 16 08 PM

Closes https://github.com/Automattic/jetpack-genesis/issues/275

## Proposed Changes

* Port the [Payment Method list implementation from Jetpack Manage](https://github.com/Automattic/wp-calypso/tree/trunk/client/jetpack-cloud/sections/partner-portal/primary/payment-methods-v2). I added a few improvements in the main container by moving all data and pagination logic within hooks to declutter some codes in the main component.
* Added temporary mock data within the `useStoredCards` hook to feed and test the component with a populated state.

## Testing Instructions
* Switch branch: `git checkout add/a4a-payment-method-list`.
* Start the server by running `yarn start-a8c-for-agencies`.
* Go to http://agencies.localhost:3000/purchases/payment-methods
* Confirm that the page functions similarly to https://cloud.jetpack.com/partner-portal/payment-methods. Note that the card data you will see is coming from mocked data.
* To test an empty state, Go to http://agencies.localhost:3000/purchases/payment-methods?flags=-a4a/mock-api-data which will have the  `a4a/mock-api-data` feature flag disabled.
* Confirm that the empty state matches the one from https://cloud.jetpack.com/partner-portal/payment-methods
## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/sections/purchases/payment-methods/payment-method-overview/index.tsx [hunk 1]
```diff
@@ -10,39 +12,108 @@ import LayoutHeader, {
 import LayoutTop from 'calypso/a8c-for-agencies/components/layout/top';
 import MobileSidebarNavigation from 'calypso/a8c-for-agencies/components/sidebar/mobile-sidebar-navigation';
 import { A4A_PAYMENT_METHODS_ADD_LINK } from 'calypso/a8c-for-agencies/components/sidebar-menu/lib/constants';
+import Pagination from 'calypso/components/pagination';
+import { PaymentMethod } from 'calypso/jetpack-cloud/sections/partner-portal/payment-methods';
+import StoredCreditCardV2 from 'calypso/jetpack-cloud/sections/partner-portal/stored-credit-card-v2';
```
Comment (jkguidaven) [inline client/a8c-for-agencies/sections/purchases/payment-methods/payment-method-overview/index.tsx:17]: To avoid bloating this PR, we are reusing the `StoredCreditCard` component from Jetpack Manage for now. This will be addressed on a separate PR.

File: client/a8c-for-agencies/sections/purchases/payment-methods/payment-method-overview/empty-state.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	const dispatch = useDispatch();
+
+	const navigateToCreateMethod = useCallback( () => {
+		dispatch( recordTracksEvent( 'calypso_a4a_license_list_empty_issue_license_click' ) );
+	}, [ dispatch ] );
+
+	return (
+		<div className="payment-method-overview-empty-state">
+			<div className="payment-method-overview-empty-state__top-content">
+				<img src={ CreditCardImg } alt="Credit Cards" />
```
Comment (cleacos) [inline client/a8c-for-agencies/sections/purchases/payment-methods/payment-method-overview/empty-state.tsx]: Should we translate the `alt` text?

File: client/a8c-for-agencies/sections/purchases/payment-methods/payment-method-overview/hooks/use-stored-cards.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import { useEffect, useState } from 'react';
+import { PaymentMethod } from 'calypso/jetpack-cloud/sections/partner-portal/payment-methods';
+
+// FIXME: Need to hook this up to the real API.
+export default function useStoredCards() {
+	const showDummyData = isEnabled( 'a4a/mock-api-data' );
+
+	const [ isFetching, setIsFetching ] = useState( showDummyData );
+
+	useEffect( () => {
```
Comment (cleacos) [inline client/a8c-for-agencies/sections/purchases/payment-methods/payment-method-overview/hooks/use-stored-cards.ts:12]: Add above `// FIXME: Don't forget to remove this fake loading delay for dummy data`

File: client/a8c-for-agencies/sections/purchases/payment-methods/payment-method-overview/style.scss [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+.payment-method-overview__stored-cards {
+	margin-block-start: 50px;
+	display: flex;
+	flex-wrap: wrap;
+	gap: 24px;
+
+	@include breakpoint-deprecated( "<660px" ) {
+		margin-block-start: 32px;
+	}
```
Comment (cleacos) [inline client/a8c-for-agencies/sections/purchases/payment-methods/payment-method-overview/style.scss]: Should we continue using this deprecated breakpoint?

I think we should take advantage of this porting from Manage to A4A to fix it wherever we can. We can use `break-small` (from `@wordpress/base-styles/breakpoints`) or `$break-small` (from `@wordpress/base-styles/mixins`) instead of it. It has a different breaking point, `600px` instead of `660px`. I don't know how it'd look.

File: client/a8c-for-agencies/sections/purchases/payment-methods/payment-method-overview/style.scss [hunk 5]
```diff
@@ -0,0 +1,110 @@
+@import "@wordpress/base-styles/breakpoints";
+@import "@wordpress/base-styles/mixins";
+
+.payment-method-overview {
+	@include breakpoint-deprecated( "<660px" ) {
```
Comment (cleacos) [inline client/a8c-for-agencies/sections/purchases/payment-methods/payment-method-overview/style.scss]: Check this breakpoint. See my comment below.

**Review by cleacos — APPROVED** [review]
It works as expected. I left some minor comments.

Other than that. LGTM!

File: client/a8c-for-agencies/sections/purchases/payment-methods/payment-method-overview/empty-state.tsx (hunk 2 above)
Comment (jkguidaven) [reply client/a8c-for-agencies/sections/purchases/payment-methods/payment-method-overview/empty-state.tsx]: Good catch 👍 this has been fixed.

File: client/a8c-for-agencies/sections/purchases/payment-methods/payment-method-overview/style.scss (hunk 4 above)
Comment (jkguidaven) [reply client/a8c-for-agencies/sections/purchases/payment-methods/payment-method-overview/style.scss]: That is a good point. This has been addressed. Thanks for the feedback. 👍

**Comment by jkguidaven** [comment]
Thanks both for the review. 🙏

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/11574875

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @jkguidaven for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/11574875).
